### ISSUE TITLE
Cannot read property 'statusCode' of undefined

### ISSUE THREAD

**Issue by technicalpickles** [byu-oit-appdev/node-duo-api#9]
Yesterday, I started seeing failures coming back from `duo-api`. I apologize in advance for sharing enough details to help debug and reproduce, but I wasn't able to capture it.

Codewise, I'm doing the usual worfklow of `duo.request 'post', '/auth/v2/preauth'` then `duo.request 'post', '/auth/v2/auth'`. I'm not sure which failed, but I got a stack trace like:

```
TypeError: Cannot read property 'statusCode' of undefined
  at Request._callback (node_modules/duo-api/lib/duo-api.js:49:71)
  at self.callback (node_modules/duo-api/node_modules/request/request.js:198:22)
  at Request.EventEmitter.emit (events.js:95:17)
  at Request.onRequestError (node_modules/duo-api/node_modules/request/request.js:820:8)
  at ClientRequest.EventEmitter.emit (events.js:95:17)
  at CleartextStream.socketErrorListener (http.js:1547:9)
  at CleartextStream.EventEmitter.emit (events.js:95:17)
  at SecurePair.<anonymous> (tls.js:1386:15)
  at SecurePair.EventEmitter.emit (events.js:95:17)
  at SecurePair.error (tls.js:1006:27)
  at EncryptedStream.CryptoStream._done (tls.js:695:22)
  at CleartextStream.read [as _read] (tls.js:496:24)
  at CleartextStream.Readable.read (_stream_readable.js:320:10)
  at EncryptedStream.onCryptoStreamFinish (tls.js:301:47)
  at EncryptedStream.g (events.js:175:14)
  at EncryptedStream.EventEmitter.emit (events.js:117:20)
  at finishMaybe (_stream_writable.js:354:12)
  at endWritable (_stream_writable.js:361:3)
  at EncryptedStream.Writable.end (_stream_writable.js:339:5)
  at EncryptedStream.CryptoStream.end (tls.js:633:31)
  at Socket.onend (_stream_readable.js:483:10)
  at Socket.g (events.js:175:14)
  at Socket.EventEmitter.emit (events.js:117:20)
  at _stream_readable.js:920:16
  at process._tickDomainCallback (node.js:459:13)
```

That line of code with statusCode undefined is https://github.com/byu-oit-appdev/node-duo-api/blob/9cb3abb4c8ae2c701fd99be9e392344a0868cc78/lib/duo-api.js#L49 . That means that somehow the conditional preceeding it isn't being matched and *somehow* `response` is undefined here. I haven't used request much, but even the readme example uses the same logic: https://github.com/request/request

I think it needs to check `error` first, and if `response` is present, report the status code.

I'm no longer seeing the error, so I think it's a case of an intermittent error from Duo causing `duo-api` to fail when reporting the failure.

**Comment by alexkolson** [comment]
Your fix is deployed in v1.0.10.

Thanks again!

**Comment by technicalpickles** [comment]
@alexkolson awesome, thanks for merging & releasing!
